### ISSUE TITLE
Math.seedrandom not documented at https://js.tensorflow.org/api/latest/

### ISSUE THREAD

**Issue by ToonTalk** [tensorflow/tfjs#2219]
The TensorFlow.js book says

> The randomness in a TensorFlow.js program can usually be disabled by calling the Math.seedrandom() function prior to creating and running the model.

I searched for such functionality in https://js.tensorflow.org/api/latest/ and there are some functions that take seeds as optional arguments but no mention of seedrandom.

Also curious why it is in Math and why no camelcasing.

**Comment by rthadur** [comment]
@ToonTalk so you want to update the document ? If yes please submit a PR.
@tafsiri do you know why camel-casing is not enforced for this ?

**Comment by nsthorat** [comment]
seedrandom() is actually this library that augments the Math object: https://github.com/davidbau/seedrandom

**Comment by ToonTalk** [comment]
[I see that](https://github.com/davidbau/seedrandom)

> Note: calling Math.seedrandom('constant') without new will make Math.random() predictable globally, which is intended to be useful for derandomizing code for testing, but should not be done in a production library. If you need a local seeded PRNG, use myrng = new Math.seedrandom('seed') instead.

Is TFJS using `Math.seedrandom('constant')` or `new Math.seedrandom('constant')`?

If the latter (which is what David Bau recommends in production code) then TFJS needs an API to replace the seed.

If it is the former then the Manning book about TFJS should at least warn people when discussing `Math.seedrandom`

**Comment by tafsiri** [comment]
After doing a quick search, each api that uses seedrandom internally allow passing in a custom seed. So I don't think it is somethign that needs to be set globally. This may be something that could be updated in the book (though it may be too late for that). 

@caiq. I'm going to set this to awaiting response in case you have any further thoughts on this, else it will auto-close in a week or two.

**Comment by mattmyne** [comment]
@tafsiri I think the advantage of Math.seedrandom over using the seed parameter for individual api calls is the former allows a predictable sequence of random numbers over multiple api calls. Using the same seed value for each api call results in identical output for each call, which is not always what's needed. Because of this I agree the use of Math.seedrandom should be in the documentation (though I'm new to TFJS so would rather not PR this myself).

**Comment by StoneT2000** [comment]
I agree @mattmyne . I'm trying to build some RL algorithms that can run on the web, but with the lack of a seed it's incredibly difficult to get reproducible results, especially in the space of RL. The python API has a global seed setter (as does pytorch) so I'm hoping there will be a way to just seed the rng globally 🙏  @tafsiri 

@tensorflowbutler

**Comment by ToonTalk** [comment]
[tf.random.set_seed](https://www.tensorflow.org/api_docs/python/tf/random/set_seed) has been in Python for a long time. Why not TFJS as well?

**Comment by mattmyne** [comment]
Perhaps this should be reopened? @google-ml-butler doesn't seem to exist anymore, who to @mention to request?

**Comment by atellmer** [comment]
I also faced the problem of initializing the seed. Here is the solution (monkey-patching):

```typescript
import seedrandom from 'seedrandom'; // tensorflow uses it

Math.random = seedrandom.alea('hello'); // hello is a seed

console.log(Math.random()); // shows a predictable result
```
